src/BackgroundTasks/BackgroundTaskManager.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace ILIAS\BackgroundTasks;
+
+/**
+ * Interface BackgroundTaskManager
+ *
+ * @package ILIAS\BackgroundTasks
+ *
+ * We add, remove or observe buckets with the background task manager.
+ *
+ */
+interface BackgroundTaskManager {
+
+	/**
+	 * @param $bucket Bucket
+	 * @param $user_ids int[]
+	 * @return Bucket
+	 */
+	public function putInQueueAndObserve($bucket, $user_ids);
```

Comment:
Yes, if the user_ids cannot be resolved to a user we need to throw an exception!